catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/QueryTransformer.java

```diff
@@ -0,0 +1,123 @@
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.codice.ddf.catalog.ui.metacard.workspace.QueryMetacardImpl;
+import org.codice.ddf.catalog.ui.query.transform.QueryBasicValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QuerySortsValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QueryValueTransformer;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryTransformer {
+
+  private EndpointUtil util;
+
+  private List<QueryValueTransformer> queryValueTransformers;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(QueryTransformer.class);
+
+  public QueryTransformer(EndpointUtil util) {
+    this.util = util;
+
+    this.queryValueTransformers =
+        Arrays.asList(new QuerySortsValueTransformer(), new QueryBasicValueTransformer());
+  }
+
+  public Map<String, Object> transformMetacardIntoMap(Metacard metacard) {
+    Map<String, Object> map = util.getMetacardMap(metacard);
+    List<Map> sorts =
+        Optional.of(map.remove("sorts"))
+            .filter(value -> value instanceof List)
+            .map(List.class::cast)
+            .map(this::mapSorts)
+            .orElse(Collections.emptyList());
+
+    Map<String, Object> filteredMap =
+        map.entrySet()
+            .stream()
+            .filter(entry -> Objects.nonNull(entry.getValue()))
+            .filter(entry -> !(entry.getValue() instanceof List) || notEmptyList(entry.getValue()))
+            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+    filteredMap.put("sorts", sorts);
+
+    return filteredMap;
+  }
+
+  private boolean notEmptyList(Object value) {
+    return value instanceof List && !((List) value).isEmpty();
+  }
+
+  public Metacard transformJsonIntoMetacard(Map<String, Object> json) {
+    QueryMetacardImpl query = new QueryMetacardImpl();
+
+    json.forEach(
+        (key, value) -> {
+          QueryValueTransformer transformer = getQueryValueTransformer(key, value);
+
+          if (transformer != null) {
+            Attribute attr = transformer.createAttribute(key, value);
+            query.setAttribute(attr);
+          } else {
+            LOGGER.trace("Appropriate query value transformer could not be found");
+          }
+        });
+
+    return query;
+  }
+
+  private List<Map> mapSorts(List<?> sortEntries) {
+    List<Map> strings =
+        sortEntries
+            .stream()
+            .filter(entry -> entry instanceof String)
+            .map(String.class::cast)
+            .map(
+                value -> {
+                  String noBrackets = value.substring(1, value.length() - 1);
+                  return noBrackets.split("=");
+                })
+            .map(parts -> Collections.singletonMap(parts[0], parts[1]))
+            .collect(Collectors.toList());
+    List<Map> maps =
```

Comment:
When is the sort a string and when it is a map?

Reply:
The sorts object is weird. It comes to the backend as a list of JSON objects however the existing backend wasn't capable of storing a list of sorts objects. It's a string when it's stored in solr and a map when it's sent to the frontend